### PR TITLE
Embedded cassandra

### PR DESCRIPTION
**Pull request by arvindyadav108**

### CODE REVIEW

File: cassandra-dac/src/main/java/org/sunbird/helper/CassandraConnectionManagerImpl.java [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+  public CassandraConnectionManagerImpl(String mode) {
+    this.mode = mode;
+  }
+
+  @Override
+  public boolean createConnection(String ip, String port, String userName, String password,
+      String keyspace) {
+    boolean connectionCreated = false;
+    if (mode.equalsIgnoreCase(JsonKey.EMBEDDED_MODE)) {
```
Comment (manzarul) [inline cassandra-dac/src/main/java/org/sunbird/helper/CassandraConnectionManagerImpl.java]: If it is embedded then return embedded object other wise return standalone object always.

File: cassandra-dac/src/main/java/org/sunbird/helper/CassandraConnectionManagerImpl.java [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+
+    Session cassandraSession = null;
+
+    PropertiesCache propertiesCache = PropertiesCache.getInstance();
+    boolean connection = false;
+
+    if (null == cassandraSessionMap.get(keyspace)) {
+      try {
+        EmbeddedCassandraServerHelper.startEmbeddedCassandra(Long.parseLong(propertiesCache.getProperty("embeddedCassandra_TimeOut")));
+        Cluster cluster = new Cluster.Builder().addContactPoints("127.0.0.1").withPort(9142)
```
Comment (manzarul) [inline cassandra-dac/src/main/java/org/sunbird/helper/CassandraConnectionManagerImpl.java]: Put ip 127.0.0.1  and port  : 9142  into external resource file or cassandra resource file.

File: cassandra-dac/src/main/java/org/sunbird/helper/CassandraConnectionManagerImpl.java [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+            .build();
+        cassandraSession = cluster.connect();
+        CQLDataLoader dataLoader = new CQLDataLoader(cassandraSession);
+        System.out.println("loading data .....");
+        dataLoader.load(new ClassPathCQLDataSet("cassandra.cql", keyspace));
+        //session = dataLoader.getSession();
+        System.out.println("EMBEDDED SESSION IS " + cassandraSession);
```
Comment (manzarul) [inline cassandra-dac/src/main/java/org/sunbird/helper/CassandraConnectionManagerImpl.java]: Never commit with  System.out.println .

File: cassandra-dac/src/main/java/org/sunbird/helper/CassandraConnectionManagerImpl.java [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+        if (null != cassandraSession) {
+          cassandraSessionMap.put(keyspace, cassandraSession);
+          cassandraclusterMap.put(keyspace, cluster);
+        }
+      } catch (TTransportException e) {
+        e.printStackTrace();
```
Comment (manzarul) [inline cassandra-dac/src/main/java/org/sunbird/helper/CassandraConnectionManagerImpl.java]: Use Logger instead of e.printStackTrace();

File: cassandra-dac/src/main/java/org/sunbird/helper/CassandraConnectionMngrFactory.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+   */
+  public static CassandraConnectionManager getObject(String name){
+
+    if(!connectionFactoryMap.containsKey(name)){
+      // create object
+      synchronized (CassandraConnectionMngrFactory.class){
+        if(name.equalsIgnoreCase(JsonKey.EMBEDDED_MODE)){
+          String mode = JsonKey.EMBEDDED_MODE;
+          if(null == connectionFactoryMap.get(JsonKey.EMBEDDED_MODE)){
+            CassandraConnectionManager embeddedcassandraConnectionManager = new CassandraConnectionManagerImpl(mode);
```
Comment (manzarul) [inline cassandra-dac/src/main/java/org/sunbird/helper/CassandraConnectionMngrFactory.java]: new CassandraConnectionManagerImpl(mode) should come from session factory .

File: cassandra-dac/src/main/java/org/sunbird/helper/CassandraConnectionManager.java [hunk 6]
```diff
@@ -1,189 +1,13 @@
...
-		  public void run() {
-			  ProjectLogger.log("started resource cleanup Cassandra.");
-			  for(Map.Entry<String,Session> entry: cassandraSessionMap.entrySet()){
-					cassandraSessionMap.get(entry.getKey()).close();
-					cassandraclusterMap.get(entry.getKey()).close();
-				}
-			  ProjectLogger.log("completed resource cleanup Cassandra.");
-		  }
-	}
+  boolean createConnection(String ip,String port,String userName,String password,String keyspace);
```
Comment (manzarul) [inline cassandra-dac/src/main/java/org/sunbird/helper/CassandraConnectionManager.java]: Write the documentation for each class label and  method label which will clearly indicate what the purpose of class and purpose of method , it's argument and return type.
